WikidataItem.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+import importer_utils as utils
+from wikidataStuff.WikidataStuff import WikidataStuff as WDS
+import pywikibot
+
+
+DATA_DIR = "data"
+
+
+class WikidataItem(object):
+    """Basic data object for upload to Wikidata."""
+
+    def make_q_item(self, qnumber):
+        """Create a regular Item."""
+        return self.wdstuff.QtoItemPage(qnumber)
+
+    def make_pywikibot_item(self, value, prop=None):
+        val_item = None
+        if type(value) is list and len(value) == 1:
+            value = value[0]
+        if utils.string_is_q_item(value):
+            val_item = self.make_q_item(value)
+        elif isinstance(value, dict) and 'quantity_value' in value:
+            number = value['quantity_value']
+            if 'unit' in value:
+                unit = self.make_q_item(value['unit'])
+            else:
+                unit = None
+            val_item = pywikibot.WbQuantity(
+                amount=number, unit=unit, site=self.repo)
+        elif value == "novalue":
+            #  to do - no_value
+            print("")
+        else:
+            val_item = value
+        return val_item
+
+    def make_statement(self, value):
+        return self.wdstuff.Statement(value)
+
+    def make_qualifier_applies_to(self, value):
+        prop_item = self.props["applies_to_part"]
+        target_item = self.wdstuff.QtoItemPage(value)
+        return self.wdstuff.Qualifier(prop_item, target_item)
+
+    def make_qualifier_startdate(self, value):
+        prop_item = self.props["start_time"]
+        value_dic = utils.date_to_dict(value, "%Y-%m-%d")
+        value_pwb = pywikibot.WbTime(year=value_dic["year"], month=value_dic[
+                                     "month"], day=value_dic["day"])
+        return self.wdstuff.Qualifier(prop_item, value_pwb)
+
+    def add_statement(self, prop_name, value, quals=None, ref=None):
+        """
+        """
+        base = self.wd_item["statements"]
+        prop = self.props[prop_name]
+        wd_claim = self.make_pywikibot_item(value, prop)
+        statement = self.make_statement(wd_claim)
+        base.append({"prop": prop,
+                     "value": statement,
+                     "quals": quals,
+                     "ref": ref})
+
+    def make_stated_in_ref(self, value, pub_date, ref_url=None):
```

Comment:
note: set this to 
januaery 20th 2017? the date when the stuff was downloaded per "last modified" of the metadata file